src/libraries/System.Net.Http/src/System/Net/Http/HttpClient.CreateDefaultHandler.Android.cs

```diff
@@ -0,0 +1,46 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Reflection;
+using System.Diagnostics.CodeAnalysis;
+
+namespace System.Net.Http
+{
+    public partial class HttpClient
+    {
+        private const string CustomHandlerTypeEnvVar = "XA_HTTP_CLIENT_HANDLER_TYPE";
+
+        private static Type? s_customHandlerType;
+
+        private static HttpMessageHandler CreateDefaultHandler()
+        {
+            if (s_customHandlerType != null)
+            {
+#pragma warning disable IL2057
+                return (HttpMessageHandler)Activator.CreateInstance(s_customHandlerType)!;
+#pragma warning restore IL2057
+            }
+
+            string? envVar = Environment.GetEnvironmentVariable(CustomHandlerTypeEnvVar)?.Trim();
+            if (!string.IsNullOrEmpty(envVar))
+            {
+#pragma warning disable IL2072
+                Type? handlerType = Type.GetType(envVar, false);
```

Comment:
I don't think using arbitrary string is good solution here. We should instantiate only the well defined XA type

Reply:
@marek-safar XA's `AndroidClientHandler` is extendable: users can subclass it, override some properies e.g. https://github.com/xamarin/xamarin-android/blob/master/src/Mono.Android/Xamarin.Android.Net/AndroidClientHandler.cs#L225-L238 here and register their subclasses via `<AndroidHttpClientHandlerType>MyAndroidClientHandler</AndroidHttpClientHandlerType>`.

Reply:
IF this is really needed then we should re-think the type injection API and provide better way than passing string during the build (how do you build a library using all these strings APIs?). This also means there can be only 1 such extension so the type can be statically set inside AndroidClientHandler.